app/services/revision_collapser.rb

```diff
@@ -0,0 +1,52 @@
+# Revisions get collapsed in three instances:
+# - A user has auto-saved: We clean previous old auto-saves
+# - A user has saved: We clean up all auto-saves
+# - A user has discarded changes: We revert the resource and remove old
+#   auto-saves
+#
+# "Collapse" is how we refer to removing multiple revisions of auto save while
+# persisting the cumulative changes. In practice it may mean we simply remove
+# old versions and persist the original state.
+#
+# Papertrail saves the *original* state of the record. When we compare
+# it in a diff, we want to compare the current state whether it's an autosave or
+# update to the original state before auto-save started. The incremental changes
+# can be discarded with out concern. Orignal <=> Current is what's important.
+# Anytime the previous revision is an auto-save we want to carry the original
+# state forward so we can always have it accessible to compare against. This
+# happens when the new revision is an update, or autosave.
+class RevisionCollapser
+  def self.call(resource)
+    return unless resource.versions.any? # Lots of specs run without versioning
+
+    last_revision = resource.versions.reorder('created_at DESC').first
```

Comment:
My understanding is that `paper_trail` already returns this in order?

https://github.com/paper-trail-gem/paper_trail#1c-basic-usage

Reply:
I'll double check this before I trust it. My gut says versions start from the first version created.
Which would be `created_at ASC`. I'll validate that and see if this needs changing.

Reply:
Paper trail default sorting is as expected `created_at ASC`.

Reply:
Isn't their example (created_at ASC, #last):

```
v = widget.versions.last
```

equivalent to yours (created_at DESC, #first)?

```
last_revision = resource.versions.reorder('created_at DESC').first 
```

Reply:
Yes. You are correct. 😆 

Change incoming.